index.d.ts

```diff
@@ -0,0 +1,59 @@
+import { VNode } from "hyperapp";
+
+/** Link */
+interface LinkProps {
+  to: string;
+  location?: Location;
+}
+export function Link(props: LinkProps): VNode<LinkProps>;
+
+/** Route */
+interface Match<P> {
+  url: string;
+  path: string;
+  isExact: boolean;
+  params: P;
+}
+interface RenderProps<P> {
+  location: Location;
+  match: Match<P>;
+}
+
+interface RouteProps<P> {
+  parent?: boolean;
+  path: string;
+  location?: Location;
+  render: (props: RenderProps<P>) => VNode<RenderProps<P>>;
+}
+
+export function Route<P>(
+  props: RouteProps<P>
+): VNode<RenderProps<P>> | undefined;
+
+/**Switch */
+export function Switch<P>(
+  props: object,
+  children: Array<VNode<RouteProps<P>>>
+): VNode<object>;
+
+/** Redirect */
+type RedirectProps = LinkProps;
```

Comment:
The `from` prop of `Redicrect` is missing in `LinkProps`

Reply:
Since `from` __or__ `location` props could be optional
https://github.com/hyperapp/router/blob/8db6747b3598c5c7832ffe1c0f09311fecb0ceee/src/Redirect.js#L4

I suggest something like:

```ts
interface LinkProps {
  to: string
  location?: Location
}

interface RedirectProps extends LinkProps {
  from?: string
}
```

Reply:
thank you it fixed.